vllm/entrypoints/chat_utils.py

```diff
@@ -1374,5 +1374,22 @@ def apply_mistral_chat_template(
             "template")
         raise ValueError(str(e)) from e
 
-def random_tool_call_id() -> str:
-    return f"chatcmpl-tool-{random_uuid()}"
+
+def get_history_tool_calls_cnt(conversation: list[ConversationMessage]):
+    idx = 0
+    for msg in conversation:
+        if msg['role'] == 'assistant':
+            tool_calls = msg.get('tool_calls')
+            if tool_calls is not None:
+                for _ in tool_calls:
+                    idx += 1
+    return idx
+
+def make_tool_call_id(id_type: str = 'random', **kwargs: Any) -> str:
+    if id_type == 'random':
+        return f"chatcmpl-tool-{random_uuid()}"
+    elif id_type == 'kimi_k2':
+        assert 'func_name' in kwargs and 'idx' in kwargs
```

Comment:
instead of assert, let's just type hint instead of kwargs. We should avoid assert as much as possible in hot path.

Reply:
ok, so we'll list every argument here, if some other model requires more arguments, they can continue to extend arguments.